src/sdk/namespace_cache.js

```diff
@@ -582,12 +629,19 @@ class NamespaceCache {
 
             const hub_stream = new stream.PassThrough();
             const hub_params = { ...params, source_stream: hub_stream };
+            const start_time = process.hrtime.bigint();
             const hub_promise = this.namespace_hub.upload_object(hub_params, object_sdk);
+            // update latency stats on 'end'
+            hub_promise.then(() => this.stats_collector.update_hub_latency_stats({
+                bucket_name: params.bucket,
+                hub_write_latency: Number(process.hrtime.bigint() - start_time) / 1e6,
+            }));
 
             const cache_finalizer = callback => hub_promise.then(() => callback(), err => callback(err));
 
             const cache_stream = new stream.PassThrough({ final: cache_finalizer });
-            const cache_params = { ...params,
+            const cache_params = {
```

Comment:
what about setting `cache_params.upload_chunks_hook` to account cache write_bytes - seems missing or something else is accounting it?